tests/js/session-tests.js

```diff
@@ -461,6 +461,25 @@ module.exports = {
         });
     },
 
+    testSyncV1() {
+        return Realm.Sync.User.register('http://localhost:9080', uuid(), 'password').then(user => {
+            return new Promise((resolve, _reject) => {
+                const config = { path: 'sync-1.x.realm', sync: { user, url: 'realm:://localhost:9080/~/sync-1.x' } };
+                try {
+                    const realm = new Realm(config);
+                }
+                catch (e) {
+                    if (e.match('/IncompatibleSyncedRealm: /')) {
+                        const backupConfig = { config: e.substring('IncompatibleSyncedRealm: '.length), sync: { user, url: 'realm:://localhost:9080/~/sync-1.x' } };
```

Comment:
Uhh, why did we switch to parsing paths out of an error message and assembling a configuration by hand? That seems incredibly ugly and error-prone. Why are we not throwing an object that exposes a pre-built configuration?

Note that the configuration that's being assembled here isn't correct: including a sync configuration will guarantee that it'll fail to open in exactly the same way that it did originally.

Reply:
Parsing strings is ugly but I don't think we can throw anything but strings (see for example https://github.com/realm/realm-js/blob/master/src/node/node_class.hpp#L309). My initial through was to throw a configuration object.